src/data/types/recordid.ts

```diff
@@ -35,16 +44,21 @@ export class RecordId<Tb extends string = string> {
 	}
 }
 
-export class StringRecordId {
+export class StringRecordId extends Value {
 	public readonly rid: string;
 
 	constructor(rid: string) {
+		super();
 		if (typeof rid !== "string")
 			throw new SurrealDbError("String Record ID must be a string");
-
 		this.rid = rid;
 	}
 
+	equals(other: unknown): boolean {
+		if (!(other instanceof StringRecordId)) return false;
+		return this.rid === other.rid;
```

Comment:
Should we then also somehow ignore whitespace here?

Reply:
Since string record ids are only used for encoding and not decoding, I would leave the responsibility for whitespace handling to the user and not perform any magic here

Reply:
Yep agree, was just trying to make a point about the whitespace haha